### PR TITLE
ISPN-11465 Upgrade to JGroups 4.2.1.Final

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-11465

* Update the Infinispan and JGroups schema versions
  in the XML configurations
* Switch to FD_ALL3 in the test JGroups stacks
* Keep using FD_ALL in the default stacks

### CODE REVIEW

File: core/src/test/resources/configs/xsite/bridge.xml [hunk 1]
```diff
@@ -29,7 +29,11 @@
            <!--break_on_coord_rsp="true"/>-->
     <MERGE3 max_interval="5000"
             min_interval="3000"/>
-    <FD_SOCK bind_addr="127.0.0.1"/>
+    <FD_SOCK/>
+    <FD_ALL3 timeout="3000"
+             interval="1000"
+    />
+    <VERIFY_SUSPECT timeout="1000"/>
```
Comment (wburns) [inline core/src/test/resources/configs/xsite/bridge.xml]: So with FD_ALL3 it suspects the member first? I thought suspecting members was tossed out of JGroups a while back?

Comment (danberindei) [reply core/src/test/resources/configs/xsite/bridge.xml]: Not sure what you mean @wburns, it was always JGroups' responsibility to suspect members. X-site backups have their own rules for taking sites offline, but a site can have 0..N site masters, all of which join the bridge cluster. 

OTOH the bridge configuration already had {{FD_SOCK}}, which suspects a node once it's stopped, and tests don't remove {{FD*}} protocols from the bridge stack based on the {{TransportFlags}}, so having {{FD_ALL3}} might just complicate debugging.

True,  we have any xsite tests using {{DISCARD}} on the site masters, so {{FD_ALL3}} and {{VERIFY_SUSPECT}} aren't really needed, but they

Comment (danberindei) [reply core/src/test/resources/configs/xsite/bridge.xml]: Removed

**Comment by wburns** [comment]
Weird I didn't see the xml parsing failures before https://ci.infinispan.org/job/Infinispan/job/PR-8052/3/#showFailuresLink

**Comment by danberindei** [comment]
There were some conflicts with another PR that updated XSD versions and I didn't check that the merged stuff really works. I fixed the failures and updated more test configurations to use `FD_ALL3`.

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Pull request merged**